mycpp/leaky_mylib.cc

```diff
@@ -132,18 +136,40 @@ bool CFileWriter::isatty() {
   return ::isatty(fileno(f_));
 }
 
+// Buf
+//
+//
+
+void Buf::Extend(Str* s) {
+  int n = len(s);
+
+  assert(cap_ >= len_);
+  if (cap_ - len_ < n + 1) {
+    cap_ = std::max(cap_ * 2, len_ + n + 1);
+    data_ = static_cast<char*>(realloc(data_, cap_));
```

Comment:
Sort of a nitpick, but I think it's easiest to reason about if `len_` and `cap_` both do NOT take into account NUL.

Simple `realloc(cap_ + 1)` in all cases

But then keep `+1` out of the `cap_ - len_` calculation, and the max calculation()

Reply:
done